pkg-manager/core/src/install/allCatalogsAreUpToDate.ts

```diff
@@ -0,0 +1,11 @@
+import { type CatalogSnapshots } from '@pnpm/lockfile-file'
+import { type Catalogs } from '@pnpm/catalogs.types'
+
+export function allCatalogsAreUpToDate (
+  catalogsConfig: Catalogs,
+  snapshot: CatalogSnapshots | undefined
+): boolean {
+  return Object.entries(snapshot ?? {})
+    .every(([catalogName, catalog]) => Object.entries(catalog ?? {})
+      .every(([alias, entry]) => entry.specifier === catalogsConfig[catalogName]?.[alias]))
+}
```

Comment:
and what if there are entries in catalogsConfig that are not present in snapshot?

Reply:
The unused catalog entries in `pnpm-workspace.yaml` won't appear at all in the `pnpm-lock.yaml` file. That felt like the preferred behavior for me, but please let me know if you see problems with that.

Reply:
That should probably be fine